### PR TITLE
[Android] pause audio playback when headphone is disconnected

### PR DESCRIPTION
**Pull request by matthuisman**
## Description

ARM7 Build:
https://jenkins.kodi.tv/job/Android-ARM/18152/artifact/kodi-20200521-96b1eeed-PR17892-merge-armeabi-v7a.apk

ARM64 Build:
https://jenkins.kodi.tv/job/Android-ARM64/14810/artifact/kodi-20200521-96b1eeed-PR17892-merge-arm64-v8a.apk

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by matthuisman** [comment]
@oldk1331
Maybe something like this?
Gave this a quick try and seems to do what you would want?
Pull out headphones and it pauses.

This is also will stop if the content can't be paused (eg. live radio stream)

Only issue is you get a bit of audio before kodi pauses / stops.
Maybe there is a way to wait for the pause / stop before continuing

I think this call in the same block: https://github.com/xbmc/xbmc/blob/ea1bbd3f7b54d5f9e5dd979b9945adab26bb6d80/xbmc/platform/android/activity/XBMCApp.cpp#L1043 is what causing Kodi to switch the audio over from headphone to speaker - so

**Comment by matthuisman** [comment]
jenkins build this please (if you listen to me)

**Comment by matthuisman** [comment]
jenkins build this please

File: xbmc/platform/android/activity/XBMCApp.cpp [hunk 1]
```diff
@@ -1022,6 +1021,13 @@ void CXBMCApp::onReceive(CJNIIntent intent)
     bool newstate = m_headsetPlugged;
     if (action == "android.intent.action.HEADSET_PLUG")
       newstate = (intent.getIntExtra("state", 0) != 0);
+
+      // If we have unplugged headset and playing content with audio then try to pause
+      if (!newstate && (m_playback_state & PLAYBACK_STATE_PLAYING) && (m_playback_state & PLAYBACK_STATE_AUDIO))
+      {
+        CApplicationMessenger::GetInstance().PostMsg(TMSG_GUI_ACTION, WINDOW_INVALID, -1,	
+                                                      static_cast<void*>(new CAction(ACTION_PAUSE)));
+      }
```
Comment (Rechi) [inline xbmc/platform/android/activity/XBMCApp.cpp:1030]: > Use spaces as tab policy with an indentation size of 2.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#32-indentation

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

File: xbmc/platform/android/activity/XBMCApp.cpp (hunk 1 above)
Comment (matthuisman) [reply xbmc/platform/android/activity/XBMCApp.cpp:1030]: I'm confused, all the indents above are 2 spaces?

Or are you talking about the linebreak on the static_cast?
That was just copied from the original merged PR
I see elsewhere in code there is no line break for the same line.
But that leads to length over 100.

Comment (fuzzard) [reply xbmc/platform/android/activity/XBMCApp.cpp:1030]: If you look at the build artifacts on jenkins at the top of the job, it always applies the clang-format as part of its build. So the diffs available will be of the clang-format changes. one for each commit, and then a whole PR single diff.

https://jenkins.kodi.tv/job/BuildMulti-PR/16106/

I find that the easiest way to "fix" these cosmetics things

Comment (matthuisman) [reply xbmc/platform/android/activity/XBMCApp.cpp:1030]: Oh, I see the issue.
there needs to be brackets after   if (action == "android.intent.action.HEADSET_PLUG")
That used to be a single line so none were required.
so without those - my indentation is indeed in-correct

**Comment by matthuisman** [comment]
jenkins build this please

**Comment by matthuisman** [comment]
jenkins build this please

**Comment by matthuisman** [comment]
jenkins build this please

**Comment by matthuisman** [comment]
jenkins build this please

**Pull request closed without merging**